widgets/common/src/NodeFilter.tsx

```diff
@@ -120,6 +132,22 @@ export default class NodeFilter extends React.Component {
         this.handleInputChange({}, event, field);
     };
 
+    optionsReducer(result: Record<string, string>, item: Partial<ResponseItem>): Record<string, string> {
+        let allowedOptions: string[] = [];
+
+        if (this.isFilteringSetFor(optionsField)) {
+            allowedOptions = this.getAllowedOptionsFor(optionsField);
+        }
+
+        if (allowedOptions.length !== 0 && !allowedOptions.includes(item.id)) {
+            return result;
+        }
```

Comment:
I've compared this logic to the previous logic and I am afraid this is not 1-1, but I don't know whether it matters :smile: 

With the new logic, if `this.getAllowedOptionsFor` returns an empty array, all items will be accepted, and with the previous logic, due to `_.intersection` with an empty array, no items would be accepted.

Thus, I believe it should be more along the lines of:

```suggestion
        if (this.isFilteringSetFor(optionsField) && !this.getAllowedOptionsFor(optionsField).includes(item.id)) {
            return result;
        }
```

meaning that we don't handle the `allowedOptions.length === 0` case in a special way

Reply:
oh... right, i missmashed the logic :(